api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/dsl/GraphTrainableModelDsl.kt

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.kotlinx.dl.api.core.dsl
+
+import org.jetbrains.kotlinx.dl.api.core.Functional
+import org.jetbrains.kotlinx.dl.api.core.GraphTrainableModel
+import org.jetbrains.kotlinx.dl.api.core.Sequential
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+
+@DslMarker
+@Retention(AnnotationRetention.SOURCE)
+internal annotation class BuilderDsl
+
+@DslMarker
+@Retention(AnnotationRetention.SOURCE)
+internal annotation class EntryDsl
+
+public class GraphTrainableModelDsl<T : GraphTrainableModel>(private val initializer: (Array<Layer>) -> T) {
+
+    private var handler: T.() -> Unit = {}
+    private var layerProvider: LayerBuilder.() -> Unit = {}
+    private var use: ((T) -> Unit)? = null
+
+    @BuilderDsl
+    public fun model(builder: T.() -> Unit) {
+        this.handler = builder
+    }
+
+    @BuilderDsl
+    public fun layers(builder: LayerBuilder.() -> Unit) {
+        this.layerProvider = builder
+    }
+
+    @BuilderDsl
+    public fun use(builder: T.() -> Unit) {
+        use = builder
+    }
+
+    public fun build(): T = initializer.invoke(LayerBuilder().apply(layerProvider).toArray()).apply(handler).apply {
```

Comment:
This line is too long and hard to read, could be reformated into several lines.

Reply:
renamed initializer createModelFunction and use invoke operator function.